src/containers/board-container.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { Light } from '../components/light.js';
+import { Board } from '../components/board.js';
+import { startingBoardOne } from '../components/test-board.js';
+
+export class BoardContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            board: [],
+            numMoves: 0,
+            hasWon: false,
+        };
+        this.board = this.getNewBoard();
+    }
+
+    /**
+     * Decides which tiles start off in the On positon.
+     * If no lights get turned on, the method will call itself again.
+     */
+    getNewBoard = () => {
+        if (this.props.isTest) return startingBoardOne;
+        let board = [];
+        let isALightOn = false;
+        for (let row = 0; row < 5; row++) {
+            board.push([]);
+            for (let col = 0; col < 5; col++) {
+                let isOn = Math.random() < 0.1 ? true : false;
+                if (isOn) isALightOn = true;
+                board[row].push(isOn);
+            }
+        }
+        if (isALightOn) {
+            return board;
+        } else {
+            return this.getNewBoard();
+        }
+    };
+
+    shouldBeOn = (row, col) => {
+        return this.board[row][col];
+    };
+
+    flip = (row, col) => {
+        this.board[row][col] = !this.board[row][col];
+    };
+
+    flipTheLights = (row, col) => {
```

Comment:
There are a bunch of functions here that just deal with data and don't necessarily have any inherent state (all of the things related to a board and updating a board from one state to the next. I would consider extracting all of the data/logic for a `board` into it's own module and to consume that module from this component. Keep the state in the component, use the functions from the board module to operate on that state. If you want a more object-oriented approach then you could create a `Board` class with data and methods which would greatly simplify the code in this component and allow it to be focused only on showing the right UI, responding to user events, and delegating figuring out the next state to some other piece of code.

Happy to expand on this if you're curious, but feel free to run with it if you'd like also.